esphome/components/sensor/filter.h

```diff
@@ -453,15 +453,20 @@ class HeartbeatFilter : public Filter, public Component {
 
 class DeltaFilter : public Filter {
  public:
-  explicit DeltaFilter(float delta, bool percentage_mode);
+  explicit DeltaFilter(float min_delta, bool min_percentage_mode, float max_delta, bool max_percentage_mode);
+
+  void set_baseline(std::function<float(float)>);
 
   optional<float> new_value(float value) override;
 
  protected:
-  float delta_;
-  float current_delta_;
+  float min_delta_;
+  bool min_percentage_mode_;
+  float max_delta_;
+  bool max_percentage_mode_;
+  optional<std::function<float(float)>> baseline_;
```

Comment:
This doesn't need to be a `std::function`, and can be pre-initialised to the default case - then there is no need to test if it's set when retrieving the baseline value, just call it.

```suggestion
  float (* baseline_)(float) = [] (float last_value) { return last_value; };
```

Reply:
Slight change in behaviour now that the current value is changed to the last value. But that's ok.

Is there a reason behind changing from std::function to a function pointer? iirc, std::function is more common within the codebase and it permits for closure state to be captured

Reply:
> Is there a reason behind changing from std::function to a function pointer?

Yes, the function pointer is lighter weight, and in this case there is nothing that needs to be captured. It's true that `std::function` is more common but in the last few months there has been quite a bit of refactoring to replace those with function pointers where possible. `std::function` is still fine where there's a capture required.

Reply:
Ok! Makes sense to me! Will change